src/main/java/org/eclipse/uprotocol/core/usubscription/v3/USubscription.java

```diff
@@ -0,0 +1,76 @@
+package org.eclipse.uprotocol.core.usubscription.v3;
+
+import org.eclipse.uprotocol.v1.UStatus;
+import org.eclipse.uprotocol.core.usubscription.v3.*;
+
+/* The following is the uSubscription API declared as an interface so it could be easily 
```

Comment:
why is this any different from all the other protobuf API's? 
not that I am against an interface, I am just curious as to why this is suddenly an interface while the others are protobuf "interface API"

Reply:
Hi @tamarafischer (and @AnotherDaniel). The reason why I added this here is because of the following reasons/issues:

1. We do not have code generators from proto yet
2. code generators create implementations not interfaces, this is problematic if I want to build generic uEs that are not dependent on an implementation (i.e. the pluggable streamer or any uE that will use uSubscription in general and not depend on say a specific  up-client implementation). 
3. USubscription has always been special in that it will not always be 1:1 mapping from the generated code, we will always have to tweak it for the specific implementation so it is better to be a language specific interface like UTransport and RpcClient. 

Above is a major shift from current PoR but we need the interface urgently to unblock everyone while we work out tooling for all the other uServices.